auth/src/main/java/io/camunda/zeebe/auth/impl/JwtAuthorizationDecoder.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.auth.impl;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import com.auth0.jwt.interfaces.Claim;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.auth0.jwt.interfaces.Verification;
+import io.camunda.zeebe.auth.api.AuthorizationDecoder;
+import io.camunda.zeebe.auth.api.JwtAuthorizationBuilder;
+import io.camunda.zeebe.util.EnsureUtil;
+import io.camunda.zeebe.util.exception.UnrecoverableException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JwtAuthorizationDecoder
+    implements JwtAuthorizationBuilder<JwtAuthorizationDecoder, Algorithm, DecodedJWT>,
+        AuthorizationDecoder<Map<String, Claim>> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JwtAuthorizationDecoder.class);
+
+  private String issuer = "zeebe-gateway";
+  private String audience = "zeebe-broker";
+  private String subject = "Authorization";
```

Comment:
>(subject): identifies the principal that is the subject of the JWT. Must be unique

🔧 While a default is hard to define, there may be better fitting alternatives than `Authorization`. First, we should attempt taking over the subject from the incoming request's authorization token, or ask Identity for the subject if it's not clear or untrusted. If that's not possible, we need a default. I suggest something like `unidentified` or `zeebe-client`